### PR TITLE
Shortcut for pausing/resuming simulation in thunderscope

### PR DESCRIPTION
**Pull request by sarphion**
### Description
You can click on the field widget, then press ctrl + space to toggle between pause and unpause state of the simulation.

### Resolved Issues
Related to #2842 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/proto/world.proto [hunk 1]
```diff
@@ -51,6 +51,11 @@ message SimulatorTick
     required double milliseconds = 1;
 }
 
+message SimulationState
+{
+    required bool is_playing = 1 [default = true];
+    // for the future, if adding a speed functionality, the proto should be added here
```
Comment (nimazareian) [inline src/proto/world.proto]: This comment can be removed.

File: src/software/thunderscope/field/world_layer.py [hunk 2]
```diff
@@ -69,6 +71,25 @@ def keyPressEvent(self, event):
         if event.key() == QtCore.Qt.Key.Key_I:
             self.display_robot_id = not self.display_robot_id
 
+        """ if ctrl and space are both pressed, pause gameplay"""
+        # if user is holding ctrl + space, send a command to simulator to pause the gameplay
```
Comment (nimazareian) [inline src/software/thunderscope/field/world_layer.py]: nit:
```suggestion
        # if user is holding ctrl + space, send a command to simulator to pause/play the gameplay
```

File: src/software/thunderscope/field/world_layer.py [hunk 3]
```diff
@@ -69,6 +71,25 @@ def keyPressEvent(self, event):
...
+            self.is_playing = not self.is_playing
+
+            self.simulator_io.send_proto(SimulationState, simulator_state)
+
+            """Register a widget to consume from a given protobuf class
+
+            :param proto_class: Class of protobuf to consume
+            :param buffer: buffer from the widget to register
+
+            """
```
Comment (nimazareian) [inline src/software/thunderscope/field/world_layer.py]: This comment can be removed. Seems to be copied from another class

File: src/software/thunderscope/thunderscope_main.py [hunk 4]
```diff
@@ -294,10 +295,24 @@ def __async_sim_ticker(tick_rate_ms):
...
             tscope.simulator_proto_unix_io.send_proto(WorldState, world_state)
 
+            # below are the changes made on saturday jan 28th, 2023
+            # example for the buffer
+            # self.primitive_set_buffer = ThreadSafeBuffer(buffer_size, PrimitiveSet)
+            simulation_state_buffer = ThreadSafeBuffer(1, SimulationState)
+            tscope.simulator_proto_unix_io.register_observer(
+                SimulationState, simulation_state_buffer
+            )
+            # or would it be tscope.simulator_proto_unix_io.register_observer(SimulationState, self.simulation_state_buffer)
```
Comment (nimazareian) [inline src/software/thunderscope/thunderscope_main.py]: Likewise, these comments

**Review by nimazareian — CHANGES_REQUESTED** [review]
Overall looks good to me! Just some nits regarding the comments

**Review by itsarune — COMMENTED** [review]
looks good, pending nima's comments

**Review by sarphion — COMMENTED** [review]
removed unnecessary comments

**Review by sauravbanna — APPROVED** [review]
looks good to me!

**Review by nimazareian — APPROVED** [review]
LGTM!

**Pull request merged**